### DISCUSSION TITLE
Error `CS8141`: Tuple element name when overriding

### DISCUSSION THREAD

**Discussion by FaustVX** [dotnet/csharplang#4921] [Q&A]
I have that code:
```csharp
interface IConverter<T1, T2>
{
  T2 Convert((T1, T2) values, T2 param);
}

class Converter : IConverter<bool, int>
{
  public int Convert((bool isCorrect, int size) values, int multiplier) // CS8141: The tuple element names in the signature of method 'Converter.Convert((bool isCorrect, int size), int)' must match the tuple element names of interface method 'IConverter<bool, int>.Convert((bool, int), int)' (including on the return type).
    => values.size * (values.isCorrect ? multiplier : 1);

/* This is what I have to use
  public int Convert((bool, int) values, int multiplier)
    => values.Item2 * (values.Item1 ? multiplier : 1);
*/
}
```

And I don't understand why this is an error.
We can rename parameter when overriding (`param` to `multiplier`).
Or assign tuple to a differently named tuple, including in Generic (`List<(bool isCorrect, int size)> list = new List<(bool, int)>()`).

For this specific usage, I can't know the name inside the Interface, but it's useful to have tuple name in the implementation (having to use `values.Item1` isn't very much readable)

So why this specific case doesn't work ?
It could be a warning, or ignored (like the other case with tuple).

**Comment by CyrusNajmabadi** [comment]
We did not feel there was good reason for the names to differ, and more often than not it indicated an unintended deviation in teh minds of most developers.

> We can rename parameter when overriding (param to multiplier).

Yes.  But this is considered not a good thing now as it just means changes in one location don't properly happen elsewhere.  It is generally discouraged.  With tuples we didn't want to make the same mistake, so we took a more firm stance on it.

**Reply by FaustVX** [reply]
Yes, but, like I said, we can assign tuple to a differently named tuple(`(int a, int b) t = (c: 1, d: 2) // CS8123: The tuple element name 'c' is ignored because a different name or no name is specified by the target type '(int a, int b)'.`), including in Generic (`List<(bool isCorrect, int size)> list = new List<(bool, int)>() // ignored`)

I understand that renaming tuple can mostly be a mistake, but I think the compiler should only warn us with CS8123.

**Reply by CyrusNajmabadi** [reply]
Those are not part of your public surface area.  Since these names are part of the public surface, we intentionally decided we would be more rigorous about ensuring that your names were not deviating.

**Reply by FaustVX** [reply]
Ok, I understand, but I still hoping for an improbable incoming warning :)
